docs/architecture/adr-104-out-of-band-state-sync.md

```diff
@@ -124,58 +138,173 @@ in order to adopt this solution.
 
 ## Decision
 
-> This section records the decision that was made.
-> It is best to record as much info as possible from the discussion that happened.
-> This aids in not having to go back to the Pull Request to get the needed information.
+As the SDK implemented a solution for all applications using it, and we do not have
+users requesting local state sync that are running in production, we will not implement the generally applicable
+solution. 
 
-State sync should support the bootstrap of new nodes from application snapshots
-obtained out-of-band by node operators.
+This ADR will outline both the solution implemented by the SDK, as well as a design proposal of a 
+generally applicable solution, that can be later on picked up and implemented in CometBFT.
 
-In other words, the following operation should be, in general terms, possible:
+## Detailed Design
 
-1. When configuring a new node, operators should be able to define where (e.g.,
-   a local file system location) the client side of state sync should look for
-   application snapshots to install or restore;
-1. Operators should be able to instruct the server side of state sync on a
-   running node to export application snapshots, when they are available,
-   to a given location (e.g., at the local file system);
-1. It is up to node operators to transfer application snapshots produced by the
-   running node (server side) to the new node (client side) to be bootstrapped
-   using this new state sync feature.
+### Application-implemented local state sync
 
+This section describes the solution to local state sync implemented by the SDK. An application can 
+chose to implement local state sync differently, or implement only a subset of the functionalities
+implemented by the SDK.
 
-## Detailed Design
+This solution exposes a command line interface enabling a node to manipulate the snapshots
+including dumping existing snapshots to an exportable format, loading and deleting exported snapshots,
+as well as bootstraping a node's state based on a local snapshot. 
```

Comment:
Node's state or application state?

To be precise here, `restore` installs a snapshot into the application. Then the bootstrap command configures the node's state and block store accordingly.